somecode.cpp

```diff
@@ -0,0 +1,32 @@
+#include <stdio.h>
+#include <string.h>
+
+int main()
+{
+   char src[40];
+   char dest[100];
+   char dest2[100];
+
+   printf("Enter your name");
+   gets(src); 
+
+   // Copy name from one place to another
+   strcpy(dest, src); 
+ 
+   int a = 10;
+   int b = 20;
+   int c = a+b;
+   printf("Result is %d",&c);
+ 
+   strncpy(dest,src, strlen(dest)); 
```

Comment:
[Banned C function detected (strncpy)](https://github.com/Microsoft/DevSkim/blob/master/guidance/DS111237.md#ruleDS111237)
strncpy is dangerous, as if the source contains 'n' or more characters, it will not null terminate the destination.

*Replacement:* Use strcpy_s or strlcpy if possible. If no safe function is viable, strcpy/strncpy should be proceeded by conditional checks to verify tha that the source string will fit in the destination with a null termnator.


Suggestions:
**[1]** +   strcpy_s(dest,  strlen(dest, src)); 
**[2]** +   strlcpy(dest, src,  strlen(dest));